README.md

```diff
@@ -102,25 +89,75 @@ cargo install --path . --force --locked --features rustfmt,cargo-fmt
 
+## Configuring Rustfmt
+
+Rustfmt is designed to be very configurable. You can create a TOML file called
+`rustfmt.toml` or `.rustfmt.toml`, place it in the project or any other parent
+directory and it will apply the options in that file. See `rustfmt
+--help=config` for the options which are available, or if you prefer to see
+visual style previews, [GitHub page](https://rust-lang.github.io/rustfmt/).
+
+By default, Rustfmt uses a style which conforms to the [Rust style guide][style
+guide] that has been formalized through the [style RFC
+process][fmt rfcs].
+
+Configuration options are either stable or unstable. Stable options can always
+be used on any channel. Unstable options are always available on nightly, but can only be used on stable and beta with an explicit opt-in (starting in Rustfmt v2.0).
+
+Unstable options are not available on stable/beta with Rustfmt v1.x.
+
+See the configuration documentation on the Rustfmt [GitHub page](https://rust-lang.github.io/rustfmt/) for details (look for the `unstable_features` section).
+
+### Rust's Editions
+
+Rustfmt is able to pick up the edition used by reading the `Cargo.toml` file if
+executed through the Cargo's formatting tool `cargo fmt`. Otherwise, the edition
+needs to be specified in `rustfmt.toml`, e.g., with `edition = "2018"`.
+
+## Limitations
+
+Rustfmt tries to work on as much Rust code as possible. Sometimes, the code
+doesn't even need to compile! As we approach a 1.0 release we are also looking
+to limit areas of instability; in particular, post-1.0, the formatting of most
+code should not change as Rustfmt improves. However, there are some things that
+Rustfmt can't do or can't do well (and thus where formatting might change
+significantly, even post-1.0). We would like to reduce the list of limitations
+over time.
 
+The following list enumerates areas where Rustfmt does not work or where the
+stability guarantees do not apply (we don't make a distinction between the two
+because in the future Rustfmt might work on code where it currently does not):
+
+* a program where any part of the program does not parse (parsing is an early
+  stage of compilation and in Rust includes macro expansion).
+* Macro declarations and uses (current status: some macro declarations and uses
+  are formatted).
+* Comments, including any AST node with a comment 'inside' (Rustfmt does not
+  currently attempt to format comments, it does format code with comments inside, but that formatting may change in the future).
+* Rust code in code blocks in comments.
+* Any fragment of a program (i.e., stability guarantees only apply to whole
+  programs, even where fragments of a program can be formatted today).
+* Code containing non-ascii unicode characters (we believe Rustfmt mostly works
+  here, but do not have the test coverage or experience to be 100% sure).
+* Bugs in Rustfmt (like any software, Rustfmt has bugs, we do not consider bug
+  fixes to break our stability guarantees).
```

Comment:
Since the main goal of this PR is to update the **running** section I think it's fine to leave this for now, however, we need to get this updated sometime soon in a follow up. The majority of this content is outdated, and although I see how it was helpful going into the initial 1.0 release, I'm not sure if it's still helpful to have a high level listing of things here within the readme.

Reply:
sounds good, im happy to update in a separate PR, want to keep diffs fairly small so reviews are faster/easier